azurerm_app_insights/main.tf

```diff
@@ -7,8 +7,9 @@ data "azurerm_resource_group" "rg" {
 # New infrastructure
 
 resource "azurerm_application_insights" "azurerm_application_insights" {
-  name                = "${local.azurerm_application_insights_name}"
+  count               = "${length(var.applications_insights)}"
+  name                = "${var.resource_name_prefix}-${var.environment}-ai-${lookup(var.applications_insights[count.index], "azurerm_application_insights_name")}"
   location            = "${data.azurerm_resource_group.rg.location}"
   resource_group_name = "${data.azurerm_resource_group.rg.name}"
-  application_type    = "${var.azurerm_application_insights_application_type}"
+  application_type    = "${lookup(var.applications_insights[count.index], "azurerm_application_insights_application_type")}"
```

Comment:
I would opt in this case to put simpler names. For example in this case I would put just "type".

Reply:
are you referring to application_type?